docs/DeveloperGuide.md

```diff
@@ -1,34 +1,315 @@
+9. If asked, choose to `Open as Project`.
+10. Click `OK` to accept the default settings.
+11. Right click on the `Duke.java` file and select `Run Duke.main()`.
+12. If the set up was done right, you should see this welcome message:
+    ````  
+     Welcome to 
+       _   _               _      _____                                _
+      | | | |   _ _       (_)    |_   _|     _ _    __ _      __      | |__     ___       _ _
+      | |_| |  | ' \      | |      | |      | '_|  / _` |    / _|     | / /    / -_)     | '_|
+       \___/   |_||_|    _|_|_    _|_|_    _|_|_   \__,_|    \__|_    |_\_\    \___|    _|_|_
+     _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""| _|"""""|
+     "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-' "`-0-0-'
+  
+    This is the list of your pinned tasks:
+  
+    Main Menu:
+    [1] Module Information
+    [2] CAP Simulator/Calculator
+    [3] Task Manager
+    [4] External Links
+    [5] Exit Program
+    ````
+
+---
+
+## 3. Design
+
+### 3.1 Architecture
+
+The `Duke` class contains the main method of our application.
+
+- At the start of the application, it initializes all the classes in sequence.
+
+We have 5 main classes for our key features.
+
+- `ModuleInfo`: contains methods for our module information features.
+- `HelpGraduation`: contains methods for our SU calculator/simulator.
+- `TaskManager`: contains methods for our task management features.
+- `Links`: contains methods for our links features.
+- `Storage`: Write data to, and loads data to hard disk
+
+![Diagram](diagrams/MainArchitecture.png)
+
+### 3.2 UI Class
+
+This class handles the interactions with the user and contains the methods for system output.
+
+---
+
+## 4. Implementation
+
+### 4.01 Add New Review
+
+This feature allows user to `add a review` for any of the modules they have taken.
+
+**NOTE**: This action will **overwrite** any existing reviews with the new review.
+
+E.g. Input:  
+`CS2113T is very fun:) I have learnt so many important concepts.`  
+`Overall 10/10 :)`  
+`/end`
+
+Output:
+
+```
+Woohoo~ Review added:  
+CS2113T is very fun:) I have learnt so many important concepts.  
+Overall 10/10 :) 
+
+Returning to module information menu...
+  ```
+Given below is the sequence diagram for the feature:  
+![Diagram](diagrams/addNewReview.png)  
+A general explanation of how this feature works:
+
+**Problem**: When the user wants to review a module, he/she should be able to type ___multiple lines
+of input___, as reviews are generally long and detailed.
+
+**Solution**: A `while` loop is used to take in multiple lines of input until the input contains
+the `/end` symbol. Anything typed after this `/end` symbol will be erased.
+
+**When overwriting this review with a new review**: The old review will be printed for user
+reference. The user will also be alerted that this action will delete his/her old review via a log.
+The log is printed out as a warning.
```

Comment:
Appreciate this form of writing due to readability and comprehension. Maybe this can be improved by mentioning the classes involved or methods involved? Similar to what was written in section 4.05.
![image](https://user-images.githubusercontent.com/77471508/113230359-6d051c80-92cb-11eb-9685-c389ee52eacb.png)